test/features/recognize.js

```diff
@@ -0,0 +1,338 @@
+
+describe("features/recognize", () => {
+  let controller;
+
+  beforeEach(async () => {
+    controller = new MockController({});
+
+    controller.bot.api.users.info
+      .withArgs({ user: "Giver" })
+      .resolves({
+        ok: true,
+        user: {
+          id: "Giver",
+          tz: "America/Los_Angeles",
+          is_bot: false,
+          is_restricted: false,
+        },
+      })
+      .withArgs({ user: "Receiver" })
+      .resolves({
+        ok: true,
+        user: {
+          id: "Receiver",
+          tz: "America/Los_Angeles",
+          is_bot: false,
+          is_restricted: false,
+        },
+      })
+      .withArgs({ user: "BotUser" })
+      .resolves({
+        ok: true,
+        user: {
+          id: "BotUser",
+          tz: "America/Los_Angeles",
+          is_bot: true,
+          is_restricted: false,
+        },
+      })
+      .withArgs({ user: "GuestUser" })
+      .resolves({
+        ok: true,
+        user: {
+          id: "GuestUser",
+          tz: "America/Los_Angeles",
+          is_bot: false,
+          is_restricted: true,
+        },
+      })
+      .withArgs({ user: "NotARealUser" })
+      .resolves({
+        ok: false,
+        error: "user_not_found",
+      });
+
+    await recognizeFeature(controller);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe("a recognition message", () => {
+    it("should update database on valid recognition", async () => {
+      const giveRecognition = sinon
+        .stub(recognition, "giveRecognition")
+        .resolves("");
+      sinon.stub(recognition, "countRecognitionsReceived").resolves(1);
+      sinon.stub(balance, "dailyGratitudeRemaining").resolves(5);
+
+      await controller.userInput({
+        text: ":fistbump: <@Receiver> Test Test Test Test Test",
+        user: "Giver",
+        channel: "SomeChannel",
+      });
+
+      expect(giveRecognition.called).to.be.true;
+    });
+
+    it("should parse out supplied message tags", async () => {
+      const giveRecognition = sinon
+        .stub(recognition, "giveRecognition")
+        .resolves("");
+      sinon.stub(recognition, "countRecognitionsReceived").resolves(1);
+      sinon.stub(balance, "dailyGratitudeRemaining").resolves(5);
+
+      await controller.userInput({
+        text: ":fistbump: <@Receiver> Test Test Test Test #Test",
+        user: "Giver",
+        channel: "SomeChannel",
+      });
+
+      expect(giveRecognition.args[0][4]).to.deep.equal(["Test"]);
+    });
+
+    it("should handle Slack API errors", async () => {
+      sinon.stub(recognition, "giveRecognition").resolves("");
+      sinon.stub(recognition, "countRecognitionsReceived").resolves(1);
+      sinon.stub(balance, "dailyGratitudeRemaining").resolves(5);
+
+      await controller.userInput({
+        text: ":fistbump: <@NotARealUser> Test Test Test Test Test",
+        user: "Giver",
+        channel: "SomeChannel",
+      });
+
+      // TODO: Test something
```

Comment:
Did you mean to leave these TODOs in here?

Reply:
Yeah, I'm not quite sure what to test at the moment, or if this might need to be restructured.

I personally want to avoid testing the response message text because I don't want unit tests to break if the wording changes, but I couldn't come up with anything else to test :|

So I just left it for now for the sake of getting the changes merged.